### PR TITLE
🔭 Fix spaces in latex input

### PR DESCRIPTION
**Pull request by rowanc1**

### CODE REVIEW

**Review by fwkoch — APPROVED** [review]
This looks good to me - it feels like the code change is pretty generic though? But I suspect that's fine - it's at least not breaking other tests.

File: packages/tex-to-myst/src/parser.ts [hunk 1]
```diff
@@ -144,6 +144,11 @@ export class TexParser implements ITexParser {
     if (!value || !this.stack.length || !('children' in top)) return;
     const last = top.children?.[top.children.length - 1];
     if (last?.type === 'text') {
+      if (last.value && value === ' ') {
+        // trim trailing whitespace and replace with a no-break space
+        last.value = last.value.replace(/\s+$/, '') + ' ';
```
[the remark this answers is not in the document]
Comment (rowanc1) [reply packages/tex-to-myst/src/parser.ts:149]: The space is a nbsp which doesn't show up in github!

[the remark this answers is not in the document]
Comment (rowanc1) [reply packages/tex-to-myst/src/parser.ts:149]: image

[the remark this answers is not in the document]
Comment (fwkoch) [reply packages/tex-to-myst/src/parser.ts:149]: Ah, of course... 👍

**Pull request merged**